tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc

```diff
@@ -398,19 +414,25 @@ Status FileSystemStoragePathSource::PollFileSystemAndInvokeCallback() {
                  << servable;
       continue;
     }
-    for (const ServableData<StoragePath>& version : versions) {
-      if (version.status().ok()) {
-        VLOG(1) << "File-system polling update: Servable:" << version.id()
-                << "; Servable path: " << version.DataOrDie()
-                << "; Polling frequency: "
-                << config_.file_system_poll_wait_seconds();
-      }
-    }
-    CallAspiredVersionsCallback(servable, versions);
+    LogVersionsAndInvokeCallback(servable, versions);
   }
   return Status::OK();
 }
 
+void FileSystemStoragePathSource::LogVersionsAndInvokeCallback(
+    const string& servable,
+    const std::vector<ServableData<StoragePath>>& versions) {
+  for (const ServableData<StoragePath> &version : versions) {
+    if (version.status().ok()) {
+      VLOG(1) << "File-system polling update: Servable:" << version.id()
+              << "; Servable path: " << version.DataOrDie()
+              << "; Polling frequency: "
+              << config_.file_system_poll_wait_seconds();
+    }
```

Comment:
Could we move the logging logic into a separate method?
It would make the code would more readable, and allow the flexibility to call CallAspiredVersionsCallback independently.

Reply:
@christisg certainly, though that seems kind of unrelated to the fix (this specific logging logic was embedded in a larger method before, and all the other logging I can see is also inline).